lib/core/test/unit/Cardano/WalletSpec.hs

```diff
@@ -122,6 +175,96 @@ walletIdInjective (walletA, walletB) = monadicIO $ liftIO $ do
     (WalletLayerFixture _ _ widsB) <- liftIO $ setupFixture walletB
     widsA `shouldNotBe` widsB
 
+walletUpdateName
+    :: (WalletId, WalletName, DummyState)
+    -> [WalletName]
+    -> Property
+walletUpdateName wallet@(_, wName0, _) names = monadicIO $ liftIO $ do
+    (WalletLayerFixture _ wl [wid]) <- liftIO $ setupFixture wallet
+    unsafeRunExceptT $ forM_ names $ \wName ->
+        updateWallet wl wid (\x -> x { name = wName })
+    wName <- fmap (name . snd) <$> unsafeRunExceptT $ readWallet wl wid
+    wName `shouldBe` last (wName0 : names)
+
+walletUpdateNameNoSuchWallet
+    :: (WalletId, WalletName, DummyState)
+    -> WalletId
+    -> WalletName
+    -> Property
+walletUpdateNameNoSuchWallet wallet@(wid', _, _) wid wName =
+    wid /= wid' ==> monadicIO $ liftIO $ do
+        (WalletLayerFixture _ wl _) <- liftIO $ setupFixture wallet
+        attempt <- runExceptT $ updateWallet wl wid (\x -> x { name = wName })
+        attempt `shouldBe` Left (ErrNoSuchWallet wid)
+
+walletUpdatePassphrase
+    :: (WalletId, WalletName, DummyState)
+    -> Passphrase "encryption-new"
+    -> Maybe (Key 'RootK XPrv, Passphrase "encryption")
+    -> Property
+walletUpdatePassphrase wallet new mxprv = monadicIO $ liftIO $ do
+    (WalletLayerFixture _ wl [wid]) <- liftIO $ setupFixture wallet
+    case mxprv of
+        Nothing -> prop_withoutPrivateKey wl wid
+        Just (xprv, pwd) -> prop_withPrivateKey wl wid (xprv, pwd)
+  where
+    prop_withoutPrivateKey wl wid = do
+        attempt <- runExceptT $ updateWalletPassphrase wl wid (coerce new, new)
+        let err = ErrUpdatePassphraseWithRootKey $ ErrWithRootKeyNoRootKey wid
+        attempt `shouldBe` Left err
+
+    prop_withPrivateKey wl wid (xprv, pwd) = do
+        unsafeRunExceptT $ attachPrivateKey wl wid (xprv, pwd)
+        attempt <- runExceptT $ updateWalletPassphrase wl wid (coerce pwd, new)
+        attempt `shouldBe` Right ()
+
+walletUpdatePassphraseWrong
+    :: (WalletId, WalletName, DummyState)
+    -> (Key 'RootK XPrv, Passphrase "encryption")
+    -> (Passphrase "encryption-old", Passphrase "encryption-new")
+    -> Property
+walletUpdatePassphraseWrong wallet (xprv, pwd) (old, new) =
+    pwd /= coerce old ==> monadicIO $ liftIO $ do
```

Comment:
Could the precondition be changed to be an assertion instead? i.e.
```haskell
attempt `shouldBe` if pwd == coerce old then Left err else Right ()
```

Same goes for the other properties with preconditions.

Reply:
Hmmm, could be. But I'd rather have two properties with two opposite pre-conditions than a single one with a variable evaluation. It makes things clearer when a test fail, what exactly is failing.